Sources/System/FilePath/FilePathParsing.swift

```diff
@@ -0,0 +1,112 @@
+/*
+ This source file is part of the Swift System open source project
+
+ Copyright (c) 2020 Apple Inc. and the Swift System project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+*/
+
+private typealias SystemChar = SystemString.Element
+
+// The separator we use internally
+private var genericSeparator: SystemChar {
+  numericCast(UInt8(ascii: "/"))
+}
+
+// The platform preferred separator
+//
+// TODO: Make private
+private var platformSeparator: SystemChar {
+#if os(Windows)
+  return numericCast(UInt8(ascii: "\\"))
+#else
+  return genericSeparator
+#endif
+}
+
+// TODO: Make private
+private func isSeparator(_ c: SystemChar) -> Bool {
+  c == genericSeparator || c == platformSeparator
+}
+
+// @available(macOS 10.16, iOS 14.0, watchOS 7.0, tvOS 14.0, *)
+extension FilePath {
+  // For invariant enforcing/checking. Should always return `nil` on
+  // a fully-formed path
+  private func _trailingSepIdx() -> Storage.Index? {
```

Comment:
```suggestion
  private func _trailingSeparatorIndex() -> Storage.Index? {
```

The abbreviation here seems ad hoc.